docker/dev/init_hook.sh

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+terminate_process () {
+    PID=$(pidof $1)
+    if [[ ${PID} != "" ]]; then
+        kill ${PID}
+        echo "process $1 terminated"
+    fi
+}
+
+if [[ "${SUPERVISOR_PROCESS_NAME}" = "agent" && "${SUPERVISOR_PROCESS_STATE}" = "terminated" ]]; then
+    terminate_process /usr/bin/vpp
```

Comment:
This part should come from the hook, e.g. `SUPERVISOR_PROCESS_PATH`. Also, the PID of the process for which the hook is called is definitely useful here as well and avoids requiring the PID lookup.